diff --git a/menu_block.module b/menu_block.module
index c91dc1d..2d666de 100644
--- a/menu_block.module
+++ b/menu_block.module
@@ -116,3 +116,19 @@ function menu_block_block_view_alter(array &$build, BlockPluginInterface $block)
     $build['#configuration']['label'] = $menuBlockTitle->label();
   }
 }
+
+/**
+ * Implements hook_module_implements_alter().
+ * @param array $implementations
+ * @param string $hook
+ *
+ * Moves the block title alter to the end of the hook stack to keep
+ * things like Tokens from stripping the label link.
+ */
+function menu_block_module_implements_alter(&$implementations, $hook) {
+  if ($hook == 'block_view_alter') {
+    $group = $implementations['menu_block'];
+    unset($implementations['menu_block']);
+    $implementations['menu_block'] = $group;
+  }
+}
diff --git a/src/Plugin/Block/MenuBlock.php b/src/Plugin/Block/MenuBlock.php
index 14ff617..00a78e7 100644
--- a/src/Plugin/Block/MenuBlock.php
+++ b/src/Plugin/Block/MenuBlock.php
@@ -130,6 +130,22 @@ class MenuBlock extends SystemMenuBlock {
       ],
     ];
 
+    $form['advanced']['label_link'] = [
+      '#type' => 'checkbox',
+      '#title' => $this->t('Link the title?'),
+      '#default_value' => $config['label_link'],
+      '#states' => [
+        'visible' => [
+          ':input[name="settings[label_type]"]' => [
+            ['value' => 'active_item'],
+            ['value' => 'parent'],
+            ['value' => 'root'],
+            ['value' => 'initial_menu_item'],
+          ],
+        ],
+      ],
+    ];
+
     $form['style'] = [
       '#type' => 'details',
       '#title' => $this->t('HTML and style options'),
@@ -184,6 +200,7 @@ class MenuBlock extends SystemMenuBlock {
     $this->configuration['parent'] = $form_state->getValue('parent');
     $this->configuration['suggestion'] = $form_state->getValue('suggestion');
     $this->configuration['label_type'] = $form_state->getValue('label_type');
+    $this->configuration['label_link'] = $form_state->getValue('label_link');
   }
 
   /**
@@ -344,6 +361,7 @@ class MenuBlock extends SystemMenuBlock {
       'expand_only_active_trails' => 0,
       'parent' => $this->getDerivativeId() . ':',
       'label_type' => 'block',
+      'label_link' => 0,
       'suggestion' => strtr($this->getDerivativeId(), '-', '_'),
     ];
   }
diff --git a/src/Title/MenuBlockActiveTrailTitle.php b/src/Title/MenuBlockActiveTrailTitle.php
index 20888ed..fc336f1 100644
--- a/src/Title/MenuBlockActiveTrailTitle.php
+++ b/src/Title/MenuBlockActiveTrailTitle.php
@@ -3,6 +3,8 @@
 namespace Drupal\menu_block\Title;
 
 use Drupal\Core\Entity\EntityRepositoryInterface;
+use Drupal\Core\Link;
+use Drupal\Core\Render\Markup;
 use Drupal\Core\Menu\MenuLinkManagerInterface;
 use Drupal\Core\Menu\MenuActiveTrailInterface;
 use Drupal\menu_block\Exception\MenuBlockNoActiveTrailFoundException;
@@ -77,7 +79,14 @@ abstract class MenuBlockActiveTrailTitle {
    */
   protected function getMenuLinkTitleByPluginId($pluginId) {
     if ($menuLink = $this->linkManager->createInstance($pluginId)) {
-      return $menuLink->getTitle();
+      $configuration = $this->menuBlock->getConfiguration();
+
+      if($configuration['label_link']) {
+        $link = Link::fromTextAndUrl($menuLink->getTitle(), $menuLink->getUrlObject())->toString();
+        return Markup::create($link);
+      } else {
+        return $menuLink->getTitle();
+      }
     }
 
     return '';
