diff --git a/src/Plugin/WebformHandler/WebformMailChimpHandler.php b/src/Plugin/WebformHandler/WebformMailChimpHandler.php
index 5c6331f..12786dc 100644
--- a/src/Plugin/WebformHandler/WebformMailChimpHandler.php
+++ b/src/Plugin/WebformHandler/WebformMailChimpHandler.php
@@ -6,7 +6,7 @@ use Drupal\Core\Config\ConfigFactoryInterface;
 use Drupal\Core\Entity\EntityTypeManagerInterface;
 use Drupal\Core\Form\FormStateInterface;
 use Drupal\Core\Serialization\Yaml;
-use Drupal\webform\WebformHandlerBase;
+use Drupal\webform\Plugin\WebformHandlerBase;
 use Drupal\webform\WebformSubmissionInterface;
 use Drupal\webform\WebformTokenManagerInterface;
 use Psr\Log\LoggerInterface;
@@ -20,8 +20,8 @@ use Symfony\Component\DependencyInjection\ContainerInterface;
  *   label = @Translation("MailChimp"),
  *   category = @Translation("MailChimp"),
  *   description = @Translation("Sends a form submission to a MailChimp list."),
- *   cardinality = \Drupal\webform\WebformHandlerInterface::CARDINALITY_UNLIMITED,
- *   results = \Drupal\webform\WebformHandlerInterface::RESULTS_PROCESSED,
+ *   cardinality = \Drupal\webform\Plugin\WebformHandlerInterface::CARDINALITY_UNLIMITED,
+ *   results = \Drupal\webform\Plugin\WebformHandlerInterface::RESULTS_PROCESSED,
  * )
  */
 class WebformMailChimpHandler extends WebformHandlerBase {